client/src/Dashboard/Dashboard.jsx

```diff
@@ -0,0 +1,140 @@
+                    maximumNumberOfUsersAllowed: createdAccount.maximumNumberOfUsersAllowed,
+                    pricePerMonth: createdAccount.pricePerMonth })
+        });
+    }
+
+    start(hub) {
+        this.setState({ hub: hub }, () => {
+            this.state.hub.start()
+
+            this.state.hub.on("AddedUser", () => {
+                this.setState({ numberOfUsers: this.state.numberOfUsers + 1 })
+            });
+        });
+    }
+
+    configureUserHub() {
+        return new signalR.HubConnectionBuilder()
+            .withUrl(`${config.apiUrl}/userHub`)
+            .build();
+    }
+
+    renderUserLimitExceededMessage() {
+        if (this.state.numberOfUsers == this.state.maximumNumberOfUsersAllowed) {
+            return (
+                <div className="alert is-error">You have exceeded the maximum number of users for your account, please upgrade your plan to increase the limit.</div>
+            );
+        }
+    }
+
+    renderUpgradeButton() {
+        if (this.state.currentPlan == 1) {
+            return (
+                <button onClick={this.upgradePlan} className="button is-success">Upgrade to Enterprise Plan</button>
+            );
+        }
+    }
+
+    upgradePlan() {
+        accountRepository.upgradePlan(this.state.accountId);
+        this.setState({ currentPlan: 2, maximumNumberOfUsersAllowed: 1000, pricePerMonth: 1000});
+    }
+
+    renderHeader() {
+        if (this.state.currentPlan == 1) {
+            return (
+                <header>Startup Plan - ${this.state.pricePerMonth}/Month</header>
+            );
+        }
+        else {
+            return (
+                <header>Enterprise Plan - ${this.state.pricePerMonth}/Month</header>
+            );
+        }
+    }
+
+    renderUpgradeSuccessMessage() {
+        if (this.state.currentPlan == 2) {
+            return (
+                <div className="alert is-success">Your account has been upgraded successfully!</div>
+            );
+        }
+    }
+
+    logout() {
+        authenticationRepository.logout()
+        this.props.history.push('/');
+    }
+    
+    render() {
+        return (
+            <div>
+                <header className="top-nav">
+                <h1>
+                    <i className="material-icons">supervised_user_circle</i>
+                    User Management Dashboard
+                </h1>
+                <button onClick={this.logout} className="button is-border">Logout</button>
+                </header>
+                
+                {this.renderUserLimitExceededMessage()}
+                {this.renderUpgradeSuccessMessage()}
+
+                <div className="plan">
+                    {this.renderHeader()}
+
+                    <div className="plan-content">
+                        <div className="progress-bar">
+                        <div style={{width: (this.state.numberOfUsers/this.state.maximumNumberOfUsersAllowed) * 100 + '%'}} className="progress-bar-usage"></div>
```

Comment:
I would not inline any logic in jsx. Instead I would declare the progress variable at the beginning of `render()`

```jsx
const { numberOfUsers, maximumNumberOfUsersAllowed } = this.state;
const progres = `${numberOfUsers/maximumNumberOfUsersAllowed * 100}%`;
```

Also the naming it too verbose, this looks shorter:

```jsx
const { userCount, userLimit } = this.state;
```

Reply:
True, they're too wordy. Changing.